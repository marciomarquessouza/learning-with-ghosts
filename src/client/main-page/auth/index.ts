export { default as logInWithEmailAndPassword } from './logInWithEmailAndPassword'
export { default as logout } from './logout'
export { default as registerWithEmailAndPassword } from './registerWithEmailAndPassword'
export { default as sendPasswordReset } from './sendPasswordReset'
export { default as signInWithGoogle } from './signInWithGoogle'
export { default as signInWithFacebook } from './signInWithFacebook'
export { default as SignInWithTwitter } from './SignInWithTwitter'
