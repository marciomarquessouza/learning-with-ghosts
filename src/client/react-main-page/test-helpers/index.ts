export * from './customRender'
export * from './mockedData'
