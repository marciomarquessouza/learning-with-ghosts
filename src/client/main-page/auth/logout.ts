import { signOut } from 'firebase/auth'
import { auth } from '../config/firebase'

function logout() {
    signOut(auth)
}

export default logout
