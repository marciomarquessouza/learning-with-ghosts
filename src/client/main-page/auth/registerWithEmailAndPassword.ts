import { createUserWithEmailAndPassword } from 'firebase/auth'
import { addDoc, collection } from 'firebase/firestore'
import { auth, db } from '../config/firebase'

async function registerWithEmailAndPassword(name: string, email: string, password: string) {
    try {
        const res = await createUserWithEmailAndPassword(auth, email, password)
        const user = res.user
        await addDoc(collection(db, 'users'), {
            uid: user.uid,
            name,
            authProvider: 'local',
            email,
        })
    } catch (err) {
        throw new Error('Error to register with email and password')
    }
}

export default registerWithEmailAndPassword
