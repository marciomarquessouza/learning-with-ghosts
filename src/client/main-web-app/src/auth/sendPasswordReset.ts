import { sendPasswordResetEmail } from 'firebase/auth'
import { auth } from '../config/firebase'

export async function sendPasswordReset(email: string) {
    try {
        await sendPasswordResetEmail(auth, email)
    } catch (err: any) {
        throw new Error('Error sending the reset e-mail')
    }
}
