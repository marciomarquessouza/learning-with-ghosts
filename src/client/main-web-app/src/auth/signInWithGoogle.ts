import { GoogleAuthProvider, signInWithPopup } from 'firebase/auth'
import { query, getDocs, collection, where, addDoc } from 'firebase/firestore'
import { auth, db } from '../config/firebase'

const googleProvider = new GoogleAuthProvider()

export async function signInWithGoogle() {
    try {
        const res = await signInWithPopup(auth, googleProvider)
        const user = res.user
        const q = query(collection(db, 'users'), where('uid', '==', user.uid))
        const docs = await getDocs(q)
        if (docs?.docs?.length === 0) {
            await addDoc(collection(db, 'users'), {
                uid: user.uid,
                name: user.displayName,
                authProvider: 'google',
                email: user.email,
            })
        }
    } catch (err) {
        throw new Error('Error to sign in with a Google Account')
    }
}
