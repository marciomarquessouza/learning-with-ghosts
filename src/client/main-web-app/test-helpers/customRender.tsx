import React from 'react'

import { RouterContext } from 'next/dist/shared/lib/router-context'
import { NextRouter } from 'next/router'
import { render, RenderOptions } from '@testing-library/react'
import { useAuthState } from 'react-firebase-hooks/auth'
import { signOut, User } from 'firebase/auth'
import AlertProvider from '../src/contexts/AlertContext'

export function createMockRouter(router: Partial<NextRouter>): NextRouter {
    return {
        basePath: '',
        pathname: '/',
        route: '/',
        query: {},
        asPath: '/',
        back: jest.fn(),
        beforePopState: jest.fn(),
        prefetch: jest.fn(),
        push: jest.fn(),
        reload: jest.fn(),
        replace: jest.fn(),
        events: {
            on: jest.fn(),
            off: jest.fn(),
            emit: jest.fn(),
        },
        isFallback: false,
        isLocaleDomain: false,
        isReady: true,
        defaultLocale: 'en',
        domainLocales: [],
        isPreview: false,
        ...router,
    }
}

interface CustomRenderOptions {
    wrappers?: RenderOptions
    auth?: {
        user?: User | null
        loading?: boolean
        signOut?: jest.Mock
    }
    alerts?: {
        openAlert?: jest.Mock
        closeAlert?: jest.Mock
    }
}

export const customRender = (
    ui: React.ReactElement,
    { wrappers, auth, alerts }: CustomRenderOptions = {}
) => {
    ;(useAuthState as jest.Mock).mockImplementation(() => [auth?.user, !!auth?.loading])
    ;(signOut as jest.Mock).mockImplementation(auth?.signOut || jest.fn())
    const router = createMockRouter({})

    return {
        router,
        ...render(
            <AlertProvider>
                <RouterContext.Provider value={router}>{ui}</RouterContext.Provider>
            </AlertProvider>,
            wrappers
        ),
    }
}
