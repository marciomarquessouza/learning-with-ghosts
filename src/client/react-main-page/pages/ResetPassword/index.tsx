import React, { useState, useEffect } from 'react'
import { useNavigate } from 'react-router-dom'

import { sendPasswordReset } from '../../auth'
import { useAlert, useAuth } from '../../hooks'
import { ALERTS_TYPE_ENUM } from '../../contexts/AlertContext'

import BackButton from '../../components/BackButton'
import ResetPasswordForm from '../../components/ResetPasswordForm'

function ResetPassword() {
    const [email, setEmail] = useState('')
    const { user, loading } = useAuth()
    const { openAlert } = useAlert()
    const navigate = useNavigate()

    const handleChangeEmail = (email: string) => {
        setEmail(email)
    }

    const handleSubmit = async () => {
        try {
            if (!email) {
                throw new Error('Please, fill in the "Email" field')
            }
            await sendPasswordReset(email)
            openAlert({
                title: 'Success!',
                message: `Check your email ${email} to reset your password`,
                type: ALERTS_TYPE_ENUM.SUCCESS,
            })
            navigate('/login')
        } catch (error: any) {
            openAlert({
                title: 'Error',
                message: error?.message,
                type: ALERTS_TYPE_ENUM.ERROR,
            })
        }
    }

    const handleRegister = () => {
        navigate('/register')
    }

    useEffect(() => {
        if (user) {
            navigate('/')
        }
    }, [user])

    return (
        <section className="h-screen bg-background font-josefin">
            <BackButton />
            <div className="px-6 h-full text-gray-800">
                <div className="flex xl:justify-center lg:justify-between justify-center items-center flex-wrap h-full g-6">
                    <div className="hidden grow-0 shrink-1 md:shrink-0 basis-auto xl:w-4/12 lg:w-6/12 md:w-9/12 mb-12 md:mb-0 md:flex items-center justify-center">
                        <img src="/img/login-logo.png" alt="Ghost Town" />
                    </div>
                    <div className="xl:ml-20 xl:w-4/12 lg:w-4/12 md:w-8/12 mb-12 md:mb-0">
                        <ResetPasswordForm
                            email={email}
                            loading={loading}
                            onChangeEmail={handleChangeEmail}
                            onClickRegister={handleRegister}
                            onSubmit={handleSubmit}
                        />
                    </div>
                </div>
            </div>
        </section>
    )
}

export default ResetPassword
