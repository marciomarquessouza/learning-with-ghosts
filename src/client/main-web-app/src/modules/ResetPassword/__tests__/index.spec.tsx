import React from 'react'
import { fireEvent, waitFor, act } from '@testing-library/react'

import { customRender } from '../../../../test-helpers'
import { sendPasswordResetEmail } from 'firebase/auth'

import ResetPassword from '..'

describe('pages >> ResetPassword', () => {
    afterEach(() => {
        jest.resetAllMocks()
    })

    it('should render the ResetPassword page properly', () => {
        const { container } = customRender(<ResetPassword />)
        expect(container).toMatchSnapshot('ResetPassword-default')
    })

    it('should navigate to /register path case the Register link is pressed', () => {
        const { getByText, router } = customRender(<ResetPassword />)

        act(() => {
            fireEvent.click(getByText('Register'))
        })

        expect(router.push).toHaveBeenCalledTimes(1)
        expect(router.push).toHaveBeenCalledWith('/register')
    })

    it('should show an error message if the Send button was pressed and the email was not filled in', async () => {
        const { getByText, container } = customRender(<ResetPassword />)

        act(() => {
            fireEvent.click(getByText('Send'))
        })

        await act(async () => {
            await waitFor(() => {
                getByText('Please enter your email first')
            })
        })

        expect(container).toMatchSnapshot('Error-email')
        expect(sendPasswordResetEmail).toHaveBeenCalledTimes(0)
    })

    it('should call the sendPasswordReset function when the "Send" button is pressed', async () => {
        const mockEmail = '__MOCK__EMAIL__'

        ;(sendPasswordResetEmail as jest.Mock).mockImplementation(() => {
            return Promise.resolve()
        })
        const { getByText, getByPlaceholderText, container } = customRender(<ResetPassword />)

        act(() => {
            fireEvent.change(getByPlaceholderText('Email address'), {
                target: { value: mockEmail },
            })
        })

        act(() => {
            fireEvent.click(getByText('Send'))
        })

        await waitFor(() => {
            getByText('Success!')
        })

        expect(container).toMatchSnapshot('success-on-sendPasswordReset')

        expect(sendPasswordResetEmail).toHaveBeenCalledTimes(1)
        expect(sendPasswordResetEmail).toHaveBeenCalledWith(undefined, mockEmail)
    })

    it('should open a error message in case of the sendPasswordReset fail', async () => {
        const mockEmail = '__MOCK__EMAIL__'
        const mockError = '__MOCK__ERROR__'

        ;(sendPasswordResetEmail as jest.Mock).mockImplementation(() => {
            throw new Error(mockError)
        })
        const { getByText, getByPlaceholderText, container } = customRender(<ResetPassword />)

        act(() => {
            fireEvent.change(getByPlaceholderText('Email address'), {
                target: { value: mockEmail },
            })
        })

        act(() => {
            fireEvent.click(getByText('Send'))
        })

        await waitFor(() => {
            getByText('Error')
        })

        expect(container).toMatchSnapshot('error-on-sendPasswordReset')
    })
})
