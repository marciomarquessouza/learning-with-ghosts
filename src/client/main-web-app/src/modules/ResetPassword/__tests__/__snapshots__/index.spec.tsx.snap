// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages >> ResetPassword should call the sendPasswordReset function when the "Send" button is pressed: success-on-sendPasswordReset 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    >
      <div
        class="pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5 transform ease-out duration-300 transition translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2"
      >
        <div
          class="p-4"
        >
          <div
            class="flex items-start"
          >
            <div
              class="flex-shrink-0"
            >
              <svg
                fill="none"
                height="15"
                viewBox="0 0 15 15"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="7.5"
                  cy="7.5"
                  fill="#34A26E"
                  r="7.5"
                />
                <path
                  d="M5.49686 9.92188L2.89061 7.31563L2.00311 8.19688L5.49686 11.6906L12.9969 4.19063L12.1156 3.30938L5.49686 9.92188Z"
                  fill="white"
                />
              </svg>
            </div>
            <div
              class="ml-3 w-0 flex-1 pt-0.5"
            >
              <p
                class="text-sm font-semibold text-primary-light"
              >
                Success!
              </p>
              <p
                class="mt-1 text-sm text-gray-500"
              >
                Check your email __MOCK__EMAIL__ to reset your password
              </p>
            </div>
            <div
              class="ml-4 flex flex-shrink-0"
            >
              <button
                class="inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Close
                </span>
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="close"
                  >
                    <path
                      d="M18.717 6.697l-1.414-1.414-5.303 5.303-5.303-5.303-1.414 1.414 5.303 5.303-5.303 5.303 1.414 1.414 5.303-5.303 5.303 5.303 1.414-1.414-5.303-5.303z"
                      id="x"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    class="h-screen bg-background font-josefin"
  >
    <section
      class="px-4 pt-4"
    >
      <a
        href="#"
      >
        <div
          class="flex flex-row items-center justify-items-center"
        >
          <svg
            class="h-10 w-10"
            fill="white"
            stroke="white"
            viewBox="0 0 219.151 219.151"
          >
            <g>
              <path
                d="M109.576,219.151c60.419,0,109.573-49.156,109.573-109.576C219.149,49.156,169.995,0,109.576,0S0.002,49.156,0.002,109.575 C0.002,169.995,49.157,219.151,109.576,219.151z M109.576,15c52.148,0,94.573,42.426,94.574,94.575 c0,52.149-42.425,94.575-94.574,94.576c-52.148-0.001-94.573-42.427-94.573-94.577C15.003,57.427,57.428,15,109.576,15z"
              />
              <path
                d="M94.861,156.507c2.929,2.928,7.678,2.927,10.606,0c2.93-2.93,2.93-7.678-0.001-10.608l-28.82-28.819l83.457-0.008 c4.142-0.001,7.499-3.358,7.499-7.502c-0.001-4.142-3.358-7.498-7.5-7.498l-83.46,0.008l28.827-28.825 c2.929-2.929,2.929-7.679,0-10.607c-1.465-1.464-3.384-2.197-5.304-2.197c-1.919,0-3.838,0.733-5.303,2.196l-41.629,41.628 c-1.407,1.406-2.197,3.313-2.197,5.303c0.001,1.99,0.791,3.896,2.198,5.305L94.861,156.507z"
              />
            </g>
          </svg>
          <span
            class="text-white text-xl ml-2"
          >
            Return
          </span>
        </div>
      </a>
    </section>
    <div
      class="px-6 h-full text-gray-800"
    >
      <div
        class="flex xl:justify-center lg:justify-between justify-center items-center flex-wrap h-full g-6"
      >
        <div
          class="hidden grow-0 shrink-1 md:shrink-0 basis-auto xl:w-4/12 lg:w-6/12 md:w-9/12 mb-12 md:mb-0 md:flex items-center justify-center"
        >
          <img
            alt="Ghost Logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="xl:ml-20 xl:w-4/12 lg:w-4/12 md:w-8/12 mb-12 md:mb-0"
        >
          <form>
            <div
              class="flex items-center my-4 before:flex-1 before:border-t before:border-gray-300 before:mt-0.5 after:flex-1 after:border-t after:border-gray-300 after:mt-0.5"
            >
              <p
                class="text-3xl text-white mb-0 mr-4"
              >
                Reset Passwoooord:
              </p>
            </div>
            <div
              class="mb-6"
            >
              <input
                class="form-control block w-full px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                name="email"
                placeholder="Email address"
                type="text"
                value="__MOCK__EMAIL__"
              />
            </div>
            <div
              class="text-center lg:text-left"
            >
              <button
                class="inline-block px-7 py-3 bg-primary-light text-white font-medium text-sm leading-snug uppercase rounded shadow-md hover:bg-primary-dark hover:shadow-lg focus:bg-primary focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
                type="submit"
              >
                Send
              </button>
              <p
                class="text-lg text-white font-semibold mt-2 pt-1 mb-0"
              >
                Don't have an account?
                 
                <a
                  class="text-primary-light hover:text-primary-dark focus:text-primary transition duration-200 ease-in-out"
                  href="#"
                >
                  Register
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`pages >> ResetPassword should open a error message in case of the sendPasswordReset fail: error-on-sendPasswordReset 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    >
      <div
        class="pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5 transform ease-out duration-300 transition translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2"
      >
        <div
          class="p-4"
        >
          <div
            class="flex items-start"
          >
            <div
              class="flex-shrink-0"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  error
                </title>
                <path
                  d="M13.728 1H6.272L1 6.272v7.456L6.272 19h7.456L19 13.728V6.272zM11 15H9v-2h2zm0-4H9V5h2z"
                  fill="#d33"
                />
              </svg>
            </div>
            <div
              class="ml-3 w-0 flex-1 pt-0.5"
            >
              <p
                class="text-sm font-semibold text-primary-light"
              >
                Error
              </p>
              <p
                class="mt-1 text-sm text-gray-500"
              >
                Error sending the reset e-mail
              </p>
            </div>
            <div
              class="ml-4 flex flex-shrink-0"
            >
              <button
                class="inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Close
                </span>
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="close"
                  >
                    <path
                      d="M18.717 6.697l-1.414-1.414-5.303 5.303-5.303-5.303-1.414 1.414 5.303 5.303-5.303 5.303 1.414 1.414 5.303-5.303 5.303 5.303 1.414-1.414-5.303-5.303z"
                      id="x"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    class="h-screen bg-background font-josefin"
  >
    <section
      class="px-4 pt-4"
    >
      <a
        href="#"
      >
        <div
          class="flex flex-row items-center justify-items-center"
        >
          <svg
            class="h-10 w-10"
            fill="white"
            stroke="white"
            viewBox="0 0 219.151 219.151"
          >
            <g>
              <path
                d="M109.576,219.151c60.419,0,109.573-49.156,109.573-109.576C219.149,49.156,169.995,0,109.576,0S0.002,49.156,0.002,109.575 C0.002,169.995,49.157,219.151,109.576,219.151z M109.576,15c52.148,0,94.573,42.426,94.574,94.575 c0,52.149-42.425,94.575-94.574,94.576c-52.148-0.001-94.573-42.427-94.573-94.577C15.003,57.427,57.428,15,109.576,15z"
              />
              <path
                d="M94.861,156.507c2.929,2.928,7.678,2.927,10.606,0c2.93-2.93,2.93-7.678-0.001-10.608l-28.82-28.819l83.457-0.008 c4.142-0.001,7.499-3.358,7.499-7.502c-0.001-4.142-3.358-7.498-7.5-7.498l-83.46,0.008l28.827-28.825 c2.929-2.929,2.929-7.679,0-10.607c-1.465-1.464-3.384-2.197-5.304-2.197c-1.919,0-3.838,0.733-5.303,2.196l-41.629,41.628 c-1.407,1.406-2.197,3.313-2.197,5.303c0.001,1.99,0.791,3.896,2.198,5.305L94.861,156.507z"
              />
            </g>
          </svg>
          <span
            class="text-white text-xl ml-2"
          >
            Return
          </span>
        </div>
      </a>
    </section>
    <div
      class="px-6 h-full text-gray-800"
    >
      <div
        class="flex xl:justify-center lg:justify-between justify-center items-center flex-wrap h-full g-6"
      >
        <div
          class="hidden grow-0 shrink-1 md:shrink-0 basis-auto xl:w-4/12 lg:w-6/12 md:w-9/12 mb-12 md:mb-0 md:flex items-center justify-center"
        >
          <img
            alt="Ghost Logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="xl:ml-20 xl:w-4/12 lg:w-4/12 md:w-8/12 mb-12 md:mb-0"
        >
          <form>
            <div
              class="flex items-center my-4 before:flex-1 before:border-t before:border-gray-300 before:mt-0.5 after:flex-1 after:border-t after:border-gray-300 after:mt-0.5"
            >
              <p
                class="text-3xl text-white mb-0 mr-4"
              >
                Reset Passwoooord:
              </p>
            </div>
            <div
              class="mb-6"
            >
              <input
                class="form-control block w-full px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                name="email"
                placeholder="Email address"
                type="text"
                value="__MOCK__EMAIL__"
              />
            </div>
            <div
              class="text-center lg:text-left"
            >
              <button
                class="inline-block px-7 py-3 bg-primary-light text-white font-medium text-sm leading-snug uppercase rounded shadow-md hover:bg-primary-dark hover:shadow-lg focus:bg-primary focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
                type="submit"
              >
                Send
              </button>
              <p
                class="text-lg text-white font-semibold mt-2 pt-1 mb-0"
              >
                Don't have an account?
                 
                <a
                  class="text-primary-light hover:text-primary-dark focus:text-primary transition duration-200 ease-in-out"
                  href="#"
                >
                  Register
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`pages >> ResetPassword should render the ResetPassword page properly: ResetPassword-default 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    />
  </div>
  <section
    class="h-screen bg-background font-josefin"
  >
    <section
      class="px-4 pt-4"
    >
      <a
        href="#"
      >
        <div
          class="flex flex-row items-center justify-items-center"
        >
          <svg
            class="h-10 w-10"
            fill="white"
            stroke="white"
            viewBox="0 0 219.151 219.151"
          >
            <g>
              <path
                d="M109.576,219.151c60.419,0,109.573-49.156,109.573-109.576C219.149,49.156,169.995,0,109.576,0S0.002,49.156,0.002,109.575 C0.002,169.995,49.157,219.151,109.576,219.151z M109.576,15c52.148,0,94.573,42.426,94.574,94.575 c0,52.149-42.425,94.575-94.574,94.576c-52.148-0.001-94.573-42.427-94.573-94.577C15.003,57.427,57.428,15,109.576,15z"
              />
              <path
                d="M94.861,156.507c2.929,2.928,7.678,2.927,10.606,0c2.93-2.93,2.93-7.678-0.001-10.608l-28.82-28.819l83.457-0.008 c4.142-0.001,7.499-3.358,7.499-7.502c-0.001-4.142-3.358-7.498-7.5-7.498l-83.46,0.008l28.827-28.825 c2.929-2.929,2.929-7.679,0-10.607c-1.465-1.464-3.384-2.197-5.304-2.197c-1.919,0-3.838,0.733-5.303,2.196l-41.629,41.628 c-1.407,1.406-2.197,3.313-2.197,5.303c0.001,1.99,0.791,3.896,2.198,5.305L94.861,156.507z"
              />
            </g>
          </svg>
          <span
            class="text-white text-xl ml-2"
          >
            Return
          </span>
        </div>
      </a>
    </section>
    <div
      class="px-6 h-full text-gray-800"
    >
      <div
        class="flex xl:justify-center lg:justify-between justify-center items-center flex-wrap h-full g-6"
      >
        <div
          class="hidden grow-0 shrink-1 md:shrink-0 basis-auto xl:w-4/12 lg:w-6/12 md:w-9/12 mb-12 md:mb-0 md:flex items-center justify-center"
        >
          <img
            alt="Ghost Logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="xl:ml-20 xl:w-4/12 lg:w-4/12 md:w-8/12 mb-12 md:mb-0"
        >
          <form>
            <div
              class="flex items-center my-4 before:flex-1 before:border-t before:border-gray-300 before:mt-0.5 after:flex-1 after:border-t after:border-gray-300 after:mt-0.5"
            >
              <p
                class="text-3xl text-white mb-0 mr-4"
              >
                Reset Passwoooord:
              </p>
            </div>
            <div
              class="mb-6"
            >
              <input
                class="form-control block w-full px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                name="email"
                placeholder="Email address"
                type="text"
                value=""
              />
            </div>
            <div
              class="text-center lg:text-left"
            >
              <button
                class="inline-block px-7 py-3 bg-primary-light text-white font-medium text-sm leading-snug uppercase rounded shadow-md hover:bg-primary-dark hover:shadow-lg focus:bg-primary focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
                type="submit"
              >
                Send
              </button>
              <p
                class="text-lg text-white font-semibold mt-2 pt-1 mb-0"
              >
                Don't have an account?
                 
                <a
                  class="text-primary-light hover:text-primary-dark focus:text-primary transition duration-200 ease-in-out"
                  href="#"
                >
                  Register
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`pages >> ResetPassword should show an error message if the Send button was pressed and the email was not filled in: Error-email 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    >
      <div
        class="pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5 transform ease-out duration-300 transition translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2"
      >
        <div
          class="p-4"
        >
          <div
            class="flex items-start"
          >
            <div
              class="flex-shrink-0"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  error
                </title>
                <path
                  d="M13.728 1H6.272L1 6.272v7.456L6.272 19h7.456L19 13.728V6.272zM11 15H9v-2h2zm0-4H9V5h2z"
                  fill="#d33"
                />
              </svg>
            </div>
            <div
              class="ml-3 w-0 flex-1 pt-0.5"
            >
              <p
                class="text-sm font-semibold text-primary-light"
              >
                Error
              </p>
              <p
                class="mt-1 text-sm text-gray-500"
              >
                Please enter your email first
              </p>
            </div>
            <div
              class="ml-4 flex flex-shrink-0"
            >
              <button
                class="inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Close
                </span>
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="close"
                  >
                    <path
                      d="M18.717 6.697l-1.414-1.414-5.303 5.303-5.303-5.303-1.414 1.414 5.303 5.303-5.303 5.303 1.414 1.414 5.303-5.303 5.303 5.303 1.414-1.414-5.303-5.303z"
                      id="x"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    class="h-screen bg-background font-josefin"
  >
    <section
      class="px-4 pt-4"
    >
      <a
        href="#"
      >
        <div
          class="flex flex-row items-center justify-items-center"
        >
          <svg
            class="h-10 w-10"
            fill="white"
            stroke="white"
            viewBox="0 0 219.151 219.151"
          >
            <g>
              <path
                d="M109.576,219.151c60.419,0,109.573-49.156,109.573-109.576C219.149,49.156,169.995,0,109.576,0S0.002,49.156,0.002,109.575 C0.002,169.995,49.157,219.151,109.576,219.151z M109.576,15c52.148,0,94.573,42.426,94.574,94.575 c0,52.149-42.425,94.575-94.574,94.576c-52.148-0.001-94.573-42.427-94.573-94.577C15.003,57.427,57.428,15,109.576,15z"
              />
              <path
                d="M94.861,156.507c2.929,2.928,7.678,2.927,10.606,0c2.93-2.93,2.93-7.678-0.001-10.608l-28.82-28.819l83.457-0.008 c4.142-0.001,7.499-3.358,7.499-7.502c-0.001-4.142-3.358-7.498-7.5-7.498l-83.46,0.008l28.827-28.825 c2.929-2.929,2.929-7.679,0-10.607c-1.465-1.464-3.384-2.197-5.304-2.197c-1.919,0-3.838,0.733-5.303,2.196l-41.629,41.628 c-1.407,1.406-2.197,3.313-2.197,5.303c0.001,1.99,0.791,3.896,2.198,5.305L94.861,156.507z"
              />
            </g>
          </svg>
          <span
            class="text-white text-xl ml-2"
          >
            Return
          </span>
        </div>
      </a>
    </section>
    <div
      class="px-6 h-full text-gray-800"
    >
      <div
        class="flex xl:justify-center lg:justify-between justify-center items-center flex-wrap h-full g-6"
      >
        <div
          class="hidden grow-0 shrink-1 md:shrink-0 basis-auto xl:w-4/12 lg:w-6/12 md:w-9/12 mb-12 md:mb-0 md:flex items-center justify-center"
        >
          <img
            alt="Ghost Logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="xl:ml-20 xl:w-4/12 lg:w-4/12 md:w-8/12 mb-12 md:mb-0"
        >
          <form>
            <div
              class="flex items-center my-4 before:flex-1 before:border-t before:border-gray-300 before:mt-0.5 after:flex-1 after:border-t after:border-gray-300 after:mt-0.5"
            >
              <p
                class="text-3xl text-white mb-0 mr-4"
              >
                Reset Passwoooord:
              </p>
            </div>
            <div
              class="mb-6"
            >
              <input
                class="form-control block w-full px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                name="email"
                placeholder="Email address"
                type="text"
                value=""
              />
            </div>
            <div
              class="text-center lg:text-left"
            >
              <button
                class="inline-block px-7 py-3 bg-primary-light text-white font-medium text-sm leading-snug uppercase rounded shadow-md hover:bg-primary-dark hover:shadow-lg focus:bg-primary focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
                type="submit"
              >
                Send
              </button>
              <p
                class="text-lg text-white font-semibold mt-2 pt-1 mb-0"
              >
                Don't have an account?
                 
                <a
                  class="text-primary-light hover:text-primary-dark focus:text-primary transition duration-200 ease-in-out"
                  href="#"
                >
                  Register
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;
